src/main/java/it/gov/pagopa/pu/sil/mapper/ClassificationsExportFileRequestMapper.java

```diff
@@ -52,13 +51,17 @@ public ClassificationsExportFileRequestDTO mapToExportFileRequest(Long organizat
   }
 
   private LocalDateIntervalFilter mapToLocalDateIntervalFilter(XMLGregorianCalendar from, XMLGregorianCalendar to) {
-    Function<XMLGregorianCalendar, LocalDate> toLocalDate = xmlGregorianCalendar ->
-      Optional.ofNullable(xmlGregorianCalendar)
-        .map(x -> x.toGregorianCalendar().toZonedDateTime().toLocalDate())
-        .orElse(null);
-    return LocalDateIntervalFilter.builder()
-      .from(toLocalDate.apply(from))
-      .to(toLocalDate.apply(to))
-      .build();
+    LocalDate localDateFrom = Optional.ofNullable(from)
+      .map(x -> x.toGregorianCalendar().toZonedDateTime().toLocalDate())
+      .orElse(null);
+    LocalDate localDateTo = Optional.ofNullable(to)
+      .map(x -> x.toGregorianCalendar().toZonedDateTime().toLocalDate())
+      .orElse(null);
+
+    if (localDateFrom != null && localDateTo != null) {
+      return LocalDateIntervalFilter.builder().from(localDateFrom).to(localDateTo).build();
+    }
```

Comment:
you are creating some optional, to else return null and then check if null...
why you don't simply check if the input is not null avoiding to create the Optional objects?

ps: is it right that you are performing the filter only if both are provided? cannot be just one of them be null and the other not?
Actually, you are ignoring a filter if the other range limit is null...
```suggestion
    LocalDate localDateFrom = Optional.ofNullable(from)
      .map(x -> x.toGregorianCalendar().toZonedDateTime().toLocalDate())
      .orElse(null);
    LocalDate localDateTo = Optional.ofNullable(to)
      .map(x -> x.toGregorianCalendar().toZonedDateTime().toLocalDate())
      .orElse(null);

    if (from != null && to != null) {
      return LocalDateIntervalFilter.builder()
         .from(from.toGregorianCalendar().toZonedDateTime().toLocalDate())
         .to(to.toGregorianCalendar().toZonedDateTime().toLocalDate()).build();
    }
```

Reply:
ps: this method is used just for legacy API, or also on the native?